odl/operator/operator.py

```diff
@@ -93,42 +77,247 @@ def _default_apply(self, x, out, *args, **kwargs):
     -------
     `None`
     """
-    out.assign(self._call(x, *args, **kwargs))
+    out.assign(op._call_out_of_place(x, **kwargs))
 
 
-class _OperatorMeta(type):
+def _signature_from_spec(func):
```

Comment:
It happens right after the fresh new instance of the class is created. Its `_call()` is inspected, and then the in-place and out-of-place methods plus the two flags are inserted by monkey-patching into the instance which is then finally returned (happens in `__new__()`). Then comes the initialization.

In fact, I think all the logic could be moved into `__init__()` now that there are no more abstract methods. I did not properly try this because it initially failed due to methods being abstract.